### ISSUE TITLE
OS singleton destroys memory pool manager too soon

### ISSUE THREAD

**Issue by rraallvv** [godotengine/godot#1083]
In those platforms that require the OS singleton to be `created/destroyed` dynamically by calling `new/delete`, the memory pool manager is destroyed before the OS singleton, causing that those objects in the singleton that were created using the memory pool manager, fail to be freed.

For instance the OS singleton in the iOS platform

```
int iphone_main(int width, int height, int argc, char** argv) {
    //...
    os = new OSIPhone(width, height);
    //...
};

void iphone_finish() {
    printf("iphone_finish\n");
    Main::cleanup(); // <-- this destroys the memory pool manager
    delete os; // ... but the manager is needed here
};
```

The error reported by Godot for this error is:

```
ERROR: free_static: Condition ' !MemoryPoolStatic::get_singleton() ' is true.
   At: /Users/rhodylugo/Desktop/Godot/core/os/memory.cpp:58.
```

**Comment by akien-mga** [comment]
What's the status on this issue? Is it still valid in the current master branch? Why was the above commit not proposed in a PR?

**Comment by akien-mga** [comment]
Can someone reproduce this issue, and if so, is the proposed patch meaningful?

**Comment by akien-mga** [comment]
Closing as old, since inactive and nobody confirmed the issue so far. Feel free to comment and ask me to reopen if the issue is still valid.

**Comment by fogelfish** [comment]
I opened a topic at the Godot Q&A that might circle back to this issue: [https://godotengine.org/qa/9630/godot-editor-interface-performance-bottleneck-on-mac-os-x?show=9697#c9697](url)

**Comment by vnen** [comment]
@fogelfish I recommend you to open another issue with your specific responsiveness problem. It's not clear that the problem you're having is related to this issue.

**Comment by vnen** [comment]
Anyway, based on the QA post, there's two confirmations that this is still valid.

**Comment by akien-mga** [comment]
Is it still valid in the current master branch?

**Comment by akien-mga** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
As there was no answer to the above request, we're closing this issue as inactive.
If you can still reproduce it, please open a new issue against a currently supported release of Godot.
